ts/test/session/unit/receiver/drop_incoming/shouldProcessMessage_test.ts

```diff
@@ -0,0 +1,130 @@
+import { describe } from 'mocha';
+import Sinon from 'sinon';
+import Long from 'long';
+import { expect } from 'chai';
+import { TestUtils } from '../../../../test-utils';
+import { shouldProcessContentMessage } from '../../../../../receiver/common';
+import { DURATION } from '../../../../../session/constants';
+
+describe('shouldProcessContentMessage', () => {
+  let envelopeTs: number;
+  beforeEach(() => {
+    TestUtils.stubWindowLog();
+    envelopeTs = Math.floor(Date.now() + Math.random() * 1000000);
+  });
+
+  afterEach(() => {
+    Sinon.restore();
+  });
+
+  describe('not a community', () => {
+    const isCommunity = false;
+    describe('with sig timestamp', () => {
+      it('if timestamps match: return true', async () => {
+        expect(
+          shouldProcessContentMessage(
+            { timestamp: envelopeTs },
+            { sigTimestamp: envelopeTs },
+            isCommunity
+          )
+        ).to.eq(true);
+      });
+      it('if timestamps do not match: return false', async () => {
+        expect(
+          shouldProcessContentMessage(
+            { timestamp: envelopeTs },
+            { sigTimestamp: envelopeTs + 2 },
+            isCommunity
+          )
+        ).to.eq(false);
+      });
+    });
+    describe('without sig timestamp', () => {
+      it('if timestamps match or not: return true', async () => {
+        expect(
+          shouldProcessContentMessage(
+            { timestamp: envelopeTs },
+            { sigTimestamp: undefined as any },
+            isCommunity
+          )
+        ).to.eq(true);
+        expect(
+          shouldProcessContentMessage({ timestamp: envelopeTs }, { sigTimestamp: 0 }, isCommunity)
+        ).to.eq(true);
+        expect(
+          shouldProcessContentMessage(
+            { timestamp: envelopeTs },
+            { sigTimestamp: Long.fromNumber(0) as any },
+            isCommunity
+          )
+        ).to.eq(true);
+      });
+    });
+  });
+
+  describe('a community', () => {
+    const isCommunity = true;
+    describe('with sig timestamp', () => {
+      it('if timestamps roughly match: return true', async () => {
+        expect(
```

Comment:
I think you should use the message paramaters here to explain when you are testing an exact match vs within 6 hours instead of just a comment. That way when we get an error we know exactly which statement failed.